src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/SerializableInfo.cs

```diff
@@ -0,0 +1,43 @@
+﻿//-----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//-----------------------------------------------------------------------------
+namespace System.Runtime.Serialization
+{
+    using System;
+    using System.Diagnostics;
+
+    public class SerializableInfo
+    {
+        public string TypeName { get; set; }
+        public string[] NonSerializedMembers { get; set; }
+
+        private SerializableInfo(string typeName, string[] nonSerializedMembers)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(typeName));
+            Debug.Assert(nonSerializedMembers != null);
+            TypeName = typeName;
+            NonSerializedMembers = nonSerializedMembers;
+        }
+
+        public static SerializableInfo Parse(string str)
+        {
+            if (string.IsNullOrEmpty(str))
+            {
+                throw new ArgumentNullException("str");
+            }
+            string[] parts = str.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
+            if (parts.Length == 0)
+            {
+                throw new ArgumentException("str");
+            }
+            string typeName = parts[0];
+            string[] nonSerializedMembers = new string[0];
+            if (parts.Length == 2)
+            {
+                nonSerializedMembers = parts[1].Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I didn't use pre-populated dictionary as I want to keep the code consistent with .NET Native. But thinking again I think the benefits outweighs the inconsistency. Updated.